Add documentation for workspace privacy (#8994)
opensearch-project/documentation-website: _dashboards/workspace/workspace-acl.md

When permission control is enabled, workspace administrators can set one of the following three access levels:

* **Private to collaborators (Default):** Only workspace collaborators can access the workspace. 
* **Anyone can view:** Grants **Read only** permissions to all workspace users, allowing them to view workspace assets. 
* **Anyone can edit:** Grants **Read and write** permissions to all users, allowing them to view, create, and update workspace assets.

Collaborators are granted higher permissions when their individual access level differs from that set in the workspace settings. For example, if workspace privacy is set to "Anyone can edit", any collaborator with read-only access will also be able to edit workspace assets.

You can set up workspace privacy on the **Create workspace** page as a **Dashboard admin**. You can also modify it on the **Collaborators** or **Workspace details** pages as a **Workspace admin** or **Dashboard admin**.

### Setting up workspace privacy during workspace creation

Use the following steps to change workspace privacy settings when creating a new workspace:

1. Choose the desired access level from the **Set up privacy** panel. 

> 2. _Optional_ Decide whether to continue configuring collaborators after workspace creation by selecting the **Go to configure the collaborators right after creating the workspace.** checkbox.

Reviewer: @natebower: "collaborator" is a workspace user. When you check the box, you'll go straight the "Collaborators" view after creating the workspace.

> 2. _Optional_ Decide whether to continue configuring collaborators after workspace creation by selecting the **Go to configure the collaborators right after creating the workspace.** checkbox.

Reviewer 2: Suggested change works for me

> 2. _Optional_ Decide whether to add collaborators after workspace creation by selecting the **Go to configure the collaborators right after creating the workspace.** checkbox.

Reviewer 2: We need to confirm that the name of the checkbox is changed to something like "Add collaborators after workspace creation".

Reviewer 2: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/9181

2. Select the **Delete x collaborators** button.
3. Review the confirmation modal that appears.
4. Select **Confirm** to remove all selected collaborators from the table.

## Configuring workspace privacy

When permission control is enabled, workspace administrators can set who has access to a workspace, with the following three access levels:

* **Private to collaborators (Default):** Gives only workspace collaborators can access the workspace. 
* **Anyone can view:** Grants **Read only** permissions to all users in the workspace, allowing them to view workspace assets. 
* **Anyone can edit:** Grants **Read and write** permissions to all users, allowing view, create, and update workspace assets.

> Collaborators are granted higher permissions when their individual access level differs from that set in the workplace settings. For example, if workspace privacy is set to “Anyone can edit”, any collaborator with read-only access level will also be able to edit the workspace assets.

Reviewer 2: ```suggestion
Collaborators are granted higher permissions when their individual access level differs from that set in the workspace settings. For example, if workspace privacy is set to "Anyone can edit", any collaborator with read-only access will also be able to edit workspace assets.
```

> 

Reviewer 2: Line 163: This is unclear as written. Is the first sentence accurate on its own? And is the second sentence a good example of the first (by default, wouldn't any collaborator have access to a workspace set to "Anyone can edit"?

Reviewer: I think you're right, the second example confuses things. It already makes sense that someone with a higher access level would have more permissions.

Alt suggestion:

```suggestion
Users at the collaborator-level are granted additional permissions even when their access level differs from that of the workspace.
```

Reviewer 2: Works for me, but removed a hyphen.

> You can set up workspace privacy on the **Create Workspace** page as a **Dashboard admin**. You can also modify it on the **Collaborators** and **Workspace details** pages, as a **Workspace admin** or **Dashboard admin**.

Reviewer 2: ```suggestion
You can set up workspace privacy on the **Create Workspace** page as a **Dashboard admin**. You can also modify it on the **Collaborators** or **Workspace details** pages as a **Workspace admin** or **Dashboard admin**.
```

> 

Reviewer 2: Confirm that the title of the "Create Workspace" page uses title case.

Reviewer: Its the other way. I adjusted it.

### Setting up workspace privacy during workspace creation

> Use the following steps to change workplace privacy settings when creating a new workspace:

Reviewer 2: ```suggestion
Use the following steps to change workspace privacy settings when creating a new workspace:
```

> 1. Choose the desired access level from the cards at the **Set up privacy** panel. 

Reviewer 2: ```suggestion
1. Choose the desired access level from the **Set up privacy** panel. 
```

> 2. _Optional_ Decide whether to continue configuring collaborators after workspace creation by selecting the **Go to configure the collaborators right after creating the workspace.** checkbox.

Reviewer 2: What do we mean by "configuring collaborators" here? The sentence is unclear. Additionally, we should use shorter and clearer text for the checkbox.

Reviewer: @Kapian1234: Where would be the best place to leave feedback on the UI text?

Reviewer 2: @Naarcha-AWS @Kapian1234 It should be something like "Add collaborators after workspace creation"

> 3. Select **Create workspace** to finish the workspace creation.

Reviewer 2: ```suggestion
3. Select **Create workspace** to create the workspace.
```

### Modifying workspace privacy on the **Collaborators** page

> Use the following steps to edit the workplace privacy settings on the **Collaborators** page:

Reviewer 2: ```suggestion
Use the following steps to edit the workspace privacy settings on the **Collaborators** page:
```

> 1. Next to **Workplace privacy**, select **Edit**. 

Reviewer 2: "Workplace" ="Workspace"?

> 1. Next to **Workplace privacy**, select **Edit**. 

Reviewer: ```suggestion
1. Next to **Workspace privacy**, select **Edit**. 
```

> 2. Select the new access level from the dropdown.

Reviewer 2: Either "list" or "menu" should follow "dropdown."

> 2. Select the new access level from the dropdown.

Reviewer: ```suggestion
2. Select the new access level from the dropdown menu.
```

3. Select **Save changes** to apply the modifications.

> ### Modifying workspace privacy on the Workspace details page

Reviewer 2: ```suggestion
### Modifying workspace privacy on the **Workspace details** page
```

> Use the following steps to edit the workplace privacy settings on the **Collaborators** page:

Reviewer 2: ```suggestion
Use the following steps to edit the workspace privacy settings on the **Workspace details** page:
```

> 1. Select the **Edit** button at the upper-right corner of the **Details** panel. 

Reviewer 2: ```suggestion
1. Select the **Edit** button in the upper-right corner of the **Details** panel. 
```

> 2. Select the new access level from the dropdown.

Reviewer 2: Either "list" or "menu" should follow "dropdown."

> 2. Select the new access level from the dropdown.

Reviewer: ```suggestion
2. Select the new access level from the dropdown menu.
```

3. Select **Save** to apply the modifications.

> 

Reviewer 2: Line 191 and subsequent is repeated text.

> ### Modifying workspace privacy on the Workspace details page

Reviewer 2: ```suggestion
```

> Use the following steps to edit the workplace privacy settings on the **Collaborators** page:

Reviewer 2: ```suggestion
```

> 1. Select the **Edit** button at the upper-right corner of the **Details** panel. 

Reviewer 2: ```suggestion
```

> 2. Select the new access level from the dropdown.

Reviewer 2: ```suggestion
```

> 3. Select **Save** to apply the modifications.

Reviewer 2: ```suggestion
```
